### DISCUSSION TITLE
Lookahead Bayesian Optimization

### DISCUSSION THREAD

**Discussion by gustavoschewinski** [meta-pytorch/botorch#2028] [General]
Hello everyone,

I'm wondering if anyone has come across an example of Lookahead Bayesian Optimization. I've attempted to utilize botorch.acquisition.multi_step_lookahead, but I haven't been able to achieve success with it yet.

Thanks! :)

**Comment by Balandat** [comment]
>  I haven't been able to achieve success with it yet.

What exactly do you mean; you don't know how to use it, or it doesn't behave as intended? 

If the former - I don't think we currently have an OSS tutorial for general multi-step lookahead BO, but maybe you can take inspiration from the test code: https://github.com/pytorch/botorch/blob/main/test/acquisition/test_multi_step_lookahead.py#L141

Also, the Knowledge Gradient is a one-step lookahead acquisition function, you can start there:  https://github.com/pytorch/botorch/blob/main/tutorials/one_shot_kg.ipynb

**Reply by gustavoschewinski** [reply]
I took inspiration from the test code, and it's functioning as expected. However, I'm uncertain about the process of retraining the models once the initial candidates have been extracted, which I already did. 

My objective is to iterate through subsequent rounds to identify the best candidates, repeating this process until I achieve the desired level of performance. And also, is it possible to add constraints for the candidates selected?

Do you know how could I do that?
